File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
#include <iostream>
#include <stack>
class Solution {
public:
    std::string reversePrefix(std::string word, char ch) {
        stack<char> st;
        int temp = -1;
        for (int i = 0; i < word.size(); i++) {
            char c = word[i];
            if (c != ch) {
                st.push(c);
            } else {
                temp = i;
                break;
            }
        }
        if (temp != -1) {
            string newStr;
            string remainingString = word.substr(temp + 1);
            newStr.push_back(ch);
            while (!st.empty()) {
                newStr.push_back(st.top());
                st.pop();
            }
        
            return newStr.append(remainingString);
        }
        else {
            return word;
        }
    }
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        Stack<Character> st = new Stack<>();
        int temp = -1;
        for (int i = 0; i < word.length(); i++) {
            char c = word.charAt(i);
            if (c != ch) {
                st.push(c);
            } else {
                temp = i;
                break;
            }
        }
        if (temp != -1) {
            StringBuilder newStr = new StringBuilder();
            String remainingString = word.substring(temp + 1);
            newStr.append(ch);
            while (!st.isEmpty()) {
                newStr.append(st.pop());
            }
            return newStr.append(remainingString).toString();
        } else {
            return word;
        }
    }
}


